### PR TITLE
Rewrite fetch to axios

### PR DESCRIPTION
**Pull request by Mechazawa**
This PR fixes Nodejs support 🎉. There are only minor differences between the browser and the nodejs api for the wrapper.

### CODE REVIEW

File: src/errors/ApiError.js [hunk 1]
```diff
@@ -35,22 +35,54 @@
...
    */
-  constructor(type, message, code, trace = null) {
+  constructor({config, request, response}) {
+    this._config = config;
+    this._response = response;
+    this._request = request;
+
+    this._code = response.status;
+
+    const {type, message, trace} = response.data.error;
```
Comment (beschoenen) [inline src/errors/ApiError.js]: Trace is not in the error anymore, it's in the root.

https://github.com/MapCreatorEU/api/blob/2a5c89b7b86af650ffbfa359d533179fa9984589/app/Macros/ResponseMacros.php#L307

Comment (beschoenen) [reply src/errors/ApiError.js]: I feel like this should be changed back, though

File: src/resources/JobResult.js [hunk 2]
```diff
@@ -139,15 +142,15 @@ export default class JobResult extends ResourceBase {
...
    *
    * @param {boolean} [value=true] - What to set the dealt-with value to
-   * @returns {Promise} - A promise that resolves with no data
    */
-  dealWith(value = true) {
+  async dealWith(value = true) {
+    value = Boolean(value);
+
     const method = value ? 'POST' : 'DELETE';
+    const url = this.url + '/deal-with';
```
Comment (beschoenen) [inline src/resources/JobResult.js]: Use string templating

File: src/utils/requests.js [hunk 3]
```diff
@@ -148,3 +150,43 @@ export function downloadFile(url, headers = {}) {
       return out;
     });
 }
+
+export function retry429ResponseInterceptor(error) {
+  if (!error.config || !error.response || error.response.status !== 429) {
+    window.z = error;
```
Comment (beschoenen) [inline src/utils/requests.js]: ?

File: src/utils/requests.js [hunk 4]
```diff
@@ -148,3 +150,43 @@ export function downloadFile(url, headers = {}) {
...
+  //
+  //   if (this.autoLogout) {
+  //     this.logout();
+  //   }
+  // }
+}
+
+export function stripAuthorizationOnRedirect() {
+
+}
```
Comment (beschoenen) [inline src/utils/requests.js]: ?

**Comment by Mechazawa** [comment]
To fix the CDN redirect issue: https://github.com/axios/axios/issues/1219

**Comment by Mechazawa** [comment]
❗️ ⚠️ `transformAxiosErrors` is broken

File: src/resources/Choropleth.js [hunk 5]
```diff
@@ -42,33 +42,24 @@ export default class Choropleth extends ResourceBase {
...
+   * @returns {Promise<DownloadedResource>} - choropleth preview
    */
-  _getDownloadHeaders() {
-    return {
-      Accept: 'application/json',
-      Authorization: this.api.auth.token.toString(),
-      'X-No-CDN-Redirect': 'true',
-    };
+  async downloadPreview() {
+    const response = await this.api.axios.get(this.url + '/preview', {
```
Comment (beschoenen) [inline src/resources/Choropleth.js]: String templating

File: src/resources/InsetMap.js [hunk 6]
```diff
@@ -39,13 +38,13 @@ export default class InsetMap extends ResourceBase {
...
-   * @returns {Promise<Object>} - choropleth json
+   * Get the inset map json
+   * @returns {Promise<Object>} - inset map json
    */
-  getJson() {
-    return fetch(this.url + '/json', {
-      headers: this._getDownloadHeaders(),
-    }).then(x => x.json());
+  async getJson() {
+    const {data} = await this.api.axios.get(this.url + '/json');
```
Comment (beschoenen) [inline src/resources/InsetMap.js]: String templating

File: src/RequestParameters.js [hunk 7]
```diff
@@ -573,6 +573,10 @@ export default class RequestParameters extends EventEmitter {
       snakeCase(x).replace(/^_/, '-'),
     ).join(',');
 
+    if (data['offset'] === 0) {
+      delete data['offset'];
+    }
```
[the remark this answers is not in the document]
Comment (Mechazawa) [reply src/RequestParameters.js]: It's left out if it's the same as the api default (0).

File: src/JobMonitor.js [hunk 8]
```diff
@@ -93,7 +94,9 @@ export default class JobMonitor {
 
   /**
    * Update the job list
-   * @returns {Promise<Number, ApiError>} - Resolves with the number of updated rows
+   * @returns {Promise<Number>} - number of updated rows
+   * @throws {ApiError}
+   * @todo refactor
```
[the remark this answers is not in the document]
Comment (Mechazawa) [reply src/JobMonitor.js]: Job monitor update logic is in need of a refactor. It's messy and uses old promise logic

**Pull request merged**